sdk/schemaregistry/schema-registry-avro/src/schemaRegistryAvroSerializer.ts

```diff
@@ -195,11 +199,20 @@ export class SchemaRegistryAvroSerializer {
       content: schema
     };
 
-    const schemaIdResponse = this.autoRegisterSchemas
-      ? await this.registry.registerSchema(description)
-      : await this.registry.getSchemaId(description);
+    let id: string;
+    if (this.autoRegisterSchemas) {
+      id = (await this.registry.registerSchema(description)).id;
+    } else {
+      const response = await this.registry.getSchemaId(description);
+      if (!response) {
+        throw new Error(
+          `Schema '${description.name}' not found in registry group '${description.group}', or not found to have matching content.`
```

Comment:
Since you asked for an alternative string, here's my best attempt, but don't feel like you have to take it. 😄 

```suggestion
          `Can't locate schema '${description.name}' in registry group '${description.group}' from given description. Did you mean to set autoRegisterSchemas?`
```

Reply:
Thanks. Elsewhere we say to avoid autoRegisterSchemas in production based on guidance from the service team. So I think we shouldn't point to it here. Leaving the message as I have it given that.